backend/SnackOverflow/src/main/java/ro/utcn/sd/vasi/SnackOverflow/controller/rest/ErrorHandler.java

```diff
@@ -0,0 +1,21 @@
+package ro.utcn.sd.vasi.SnackOverflow.controller.rest;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import ro.utcn.sd.vasi.SnackOverflow.dto.ErrorDTO;
+
+@Component
+@RestControllerAdvice
+public class ErrorHandler {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(Exception.class)
+    //ce se intampla daca ex nu e de tipu clasei din ExceptionHandler?
+    public ErrorDTO handlerError(Exception ex) {
```

Comment:
Only the exceptions of the type(s) in the `@ExceptionHandler` are sent to this method. You could have multiple such methods (one for each big exception type). If there is a mismatch between the parameter type from your method and the class in the annotation, you might get class cast exceptions at runtime...